warcprox/trough.py

```diff
@@ -0,0 +1,238 @@
+'''
```

Comment:
This should be de-duped against the code in the trough repo and the ait5 repo, at some point. We shouldn't have 3 copies of essentially the same code floating around. This seems like the most promising of the three starting points.